src/components/profile-edit.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from "react";
+
+class ProfileEdit extends Component {
+	constructor() {
+		super();
+
+		this.state = {
+			viewMode: true,
+			value: ""
+		};
+	}
+
+	get editItem() {
+		return event => {
+			event.preventDefault();
+
+			this.setState( {
+				viewMode: false
+			} );
+		};
+	}
+
+	get changeValue() {
+		return event => this.setState( {
+			value: event.target.value
+		} );
+	}
+
+	get saveItem() {
+		return event => {
+			event.preventDefault();
+
+			this.props.onSave( this.state.value );
+
+			this.setState( {
+				viewMode: true
+			} );
+		};
+	}
+
+	get cancelEdit() {
+		return () => this.setState( {
+			value: this.props.value,
+			viewMode: true
+		} );
+	}
+
+	componentWillMount() {
+		this.setState( {
+			value: this.props.value
```

Comment:
the input value is connected to this state, and I'm avoiding changes on the props.